[IMP] POS: update preparation display page (#17474)
odoo/documentation: content/applications/sales/point_of_sale/extra/preparation.rst

Preparation display
===================

The preparation display feature enables efficient management of POS orders that require preparation

> prior to completion. It is intended for use in environments where ordered items must be assembled,

Reviewer: ```suggestion
The preparation display helps manage POS orders that require preparation
prior to completion. It is intended for use in environments where ordered items must be assembled,
```
or something similar but:
- this sounded a bit too formal for our usual style IMO
- I don't think we need to add *feature*

cooked, or otherwise processed before they are handed over to the customer.

.. admonition:: Use cases

   - **For retail**: Once an order is completed and paid in the :ref:`POS register
     <pos/use/open-register>`, it is sent to the preparation display. This notifies the preparation
     team to collect and prepare the purchased items for customer pickup.
   - **For restaurants**: Once orders are placed in the :ref:`POS register <pos/use/open-register>`,
     they are sent to the preparation display to inform the kitchen about the meals that need to be
     prepared.

.. _pos/preparation/configuration:

Configuration
=============

To create and set up a preparation display, go to :menuselection:`Point of Sale --> Orders -->

> Preparation Display`, and click :guilabel:`New`. Then, configure it:

Reviewer: ```suggestion
Preparation Display`, click :guilabel:`New`, and configure it:
```

- Enter a descriptive :guilabel:`Name` (e.g., `Main Kitchen`, `Bar`, `Pickup zone`).
- Select the :guilabel:`Point of Sale` that sends orders to this display. Leave it empty to link it

>   to *all* points of sale.

Reviewer: ```suggestion
- Select the :guilabel:`Point of Sale` that sends orders to this display. Leave the field empty to link it
  to *all* points of sale.
```

- Specify the POS :ref:`product categories <pos/products/categories>` to be sent to this display.

>   Leave it empty to allow *all* categories.

Reviewer: ```suggestion
  Leave the field empty to allow *all* categories.
```

- Enable the :guilabel:`Auto clear` option to automatically remove orders from the :ref:`Order

>   Status Screen <preparation/order-status-screen>` after a given time.

Reviewer: ```suggestion
- Enable the :guilabel:`Auto clear` option to automatically remove orders from the :ref:`order
  status screen <preparation/order-status-screen>` after a given time.
```

- Define the steps required for orders to be processed in the :guilabel:`Stages` tab:

  - Click :guilabel:`Add a line` to add a stage.
  - Optionally, assign colors to stages for better visibility.

>   - Set an :guilabel:`Alert timer (min)` for each stage to define the expected processing

Reviewer: maybe a nitpick but I would consider these to be steps so I'd use a numbered list for this sub-list here?

Author: Sure, why not 😃

    time. Orders are highlighted when this time is exceeded.

- If the :guilabel:`POS Enterprise SMS Whatsapp` module is installed, activate :guilabel:`SMS
  Enabled` and/or :guilabel:`WhatsApp Enabled` in the :guilabel:`Communication` tab to notify
  customers. Then, select templates for :guilabel:`Order received` and :guilabel:`Order is

>   ready`.

Reviewer: since this doesn't seem to be installed by default, I think i would maybe put this in a tip?

.. note::
   To edit an existing preparation display, click :icon:`fa-ellipsis-v` :guilabel:`Dropdown

>    menu` on the display's card and select :guilabel:`Configure`.

Reviewer: And I'd also make this a tip

.. _pos/preparation/application:

Preparation display overview
============================

> To view all preparation displays, and related :ref:`order status screens

Reviewer: ```suggestion
To view all preparation displays and their related :ref:`order status screens
```

<preparation/order-status-screen>`, go to :menuselection:`Point of Sale --> Orders --> Preparation
Display`.

.. image:: preparation/display-card.png
   :alt: Kanban view of the preparation display
   :scale: 85 %

The display card shows:

- The configured stages, e.g., `To prepare`, `Ready`, `Completed`.
- The number of orders currently :guilabel:`In progress`.
- The :guilabel:`Average time` employees usually take to complete an order.

> - The :ref:`Order Status Screen <preparation/order-status-screen>` button, if installed.

Reviewer: I'm not really sure what you mean by *if installed* here, could you please clarify?

Author: It is possible to uninstall the Order Status Screen. It is installed by default but can be uninstalled. Maybe I gave to much information here....

Reviewer: Oh really! I'd be interested to maybe discuss this further if that's ok?

.. tip::
   - A default preparation display named :guilabel:`Kitchen Display` is automatically configured
     when creating a new :doc:`restaurant point of sale <../restaurant>` :ref:`through the
     onboarding screen <pos/use/create-pos>` and selecting the :guilabel:`Restaurant` card.
   - The **Kitchen Display** app is automatically installed whenever a :doc:`restaurant point of
     sale <../restaurant>` is :ref:`created <pos/use/create-pos>` and provides quick access to the

>      :guilabel:`Preparation Display` page from the main dashboard.

Reviewer: ```suggestion
     :guilabel:`Preparation Display` model from the main dashboard.
```
I wanted to use *view* but it's not technically correct, but I fear *model* might be a bit too technical. I wouldn't use *page* though because, maybe that's just me, but it makes it sound like this is outside of odoo (at least that's what I understood before I actually opened the Kitchen Display app and realized this was the exact same as going through the POS menu 🙈 ). Any thoughts or ideas?

Author: Maybe just "quick access to all preparation displays configured"? OR "quick access to the overview of all preparation displays"? Or "to the cards overview of all preparation displays"

Reviewer: What about mixing your suggestions into: *quick access to the preparation display overview*?

.. note::
   To use the preparation display, log in via a web browser on a mobile device or a touchscreen

>    connected via USB or HDMI. IoT boxes are not supported.

Reviewer: Thank you for adding this note, great idea to clarify this here!

.. _pos/preparation/application/use:

Using the preparation display
-----------------------------

To open the preparation display, click :guilabel:`Preparation Screen` on the display card. The

> employee interface consists of a :ref:`top bar <pos/preparation/top-bar>`, a :ref:`side panel

Reviewer: ```suggestion
To open the preparation display, click :guilabel:`Preparation Screen` on the Kanban card. The
interface consists of a :ref:`top bar <pos/preparation/top-bar>`, a collapsible :ref:`side panel
```
- usual term for this UI element
- in case people wonder why they're not seeing the side panel (I know you mention that in the related section but it doesn't hurt to have it here as well?)
- "employee interface" sounds a bit odd to me and I think it's not really needed (I do understand why you added it though so let's discuss this if you disagree!)

<pos/preparation/sidepanel>`, and the :ref:`order cards <pos/preparation/order-cards>` processing

> section.

Reviewer: ```suggestion
<pos/preparation/sidepanel>`, and the :ref:`order cards <pos/preparation/order-cards>` section.
```
or

```suggestion
<pos/preparation/sidepanel>`, and the :ref:`order processing <pos/preparation/order-cards>` section.
```
but I feel like both is a bit redundant/overkill?

.. _pos/preparation/top-bar:

Top bar
~~~~~~~

The top bar shows the progress of :guilabel:`All` orders across configured stages, such as
`To prepare`/`To cook`, `Ready`, and `Completed`, along with the number of orders per stage. Click a

> stage to see its orders.

Reviewer: ```suggestion
stage to see the related orders.
```
(might be nitpick)

.. image:: preparation/order-count.png
   :alt: Order count in the top bar.

> Click the :icon:`fa-undo` :guilabel:`Recall` button to reset the last order stage move, and use the

Reviewer: ```suggestion
Click the :icon:`fa-undo` :guilabel:`Recall` button to undo the last order stage move, and use the
```
- but then I'm not sure it's 100% clear that this is only for the selected/stage: e.g., if the "last order stage move" was on the To prepare stage, but right now I'm on the Ready stage, the recall button is greyed out. What do you think?
- I think we also need to mention the fact that for the last stage, there's a Done button instead of a Recall button?

Author: Good catch! I will add this information ❤️

:icon:`fa-sign-out` :guilabel:`Close` button to exit the display.

.. _pos/preparation/sidepanel:

Side panel
~~~~~~~~~~

> The side panel is folded by default. Open it by clicking the icon in the upper-left corner. Then,

Reviewer: We could use the [Odoo UI icon](https://www.odoo.com/documentation/master/developer/reference/user_interface/icons.html) oi-panel-right here; I know it's reversed, but it's close enough?

Author: Yes, I saw this one as well and even tried to create a new icon (lol), but then removed it because it is not correct. Do you think it is better to use this similar icon than no icon?

Reviewer: I suppose this is really subjective but I would say yes maybe?

filter orders by:

- :guilabel:`Time`, such as :guilabel:`Today` or :guilabel:`Tomorrow`

> - :doc:`Preset <presets>`, such as :guilabel:`Dine in` or :guilabel:`Delivery`

Reviewer: ```suggestion
- :doc:`Preset <presets>`, such as :guilabel:`Dine in` or :guilabel:`Delivery`, if applicable
```
since this wouldn't be displayed if no presets are configured, right?

- :ref:`POS categories <pos/products/categories>` defined on the :ref:`display form
  <pos/preparation/configuration>`, such as :guilabel:`Food` or :guilabel:`Drinks`

The total number of orders/category items is displayed next to the corresponding filter. Each
filter section consists of a breakdown of the individual elements.

Use the :icon:`fa-search-minus` and :icon:`fa-search-plus` zoom control at the bottom of the side
panel to adjust the card view size.

> 

Reviewer: And should we mention the Clear All Orders button as well?

<img width="246" height="289" alt="Image" src="https://github.com/user-attachments/assets/95f0a34c-cb28-45ac-9ae5-b166b8bc1e85" />

Author: Good catch! I did miss this indeed. Thank you :) I will add it

.. _pos/preparation/order-cards:

Order cards
~~~~~~~~~~~

> Order cards represent individual orders and contain relevant information, including:

Reviewer: Should we mention the stage here, for good measure?

```suggestion
Order cards represent individual orders in the selected stage and display the following information:
```
EDIT: but I know it could also be "All" orders so I'm not sure ... see my related comment on line 137

Author: I wouldn't mention the stage here, as the order cards under "All" is the most "complete" and I would then rather mention this detail on line 137

Reviewer: Ok then, let's do that, thank you!

- Associated :ref:`tables <pos/restaurant/floors>`, order numbers, :ref:`customer
  <pos/use/customers>` or order names.
- Number of guests.

> - :doc:`Preset <presets>` service types, if configured.

Reviewer: I know the default presets (and most common use of presets currently, by far) is service types but technically, presets could be anything so should we stay a bit more general?

```suggestion
- :doc:`Presets <presets>`, if configured.
```

> - The selected time and date, if :ref:`configured <pos/presets/preset-form>`.

Reviewer: ```suggestion
- The selected time and date, if :ref:`orders are managed by time for the preset <pos/presets/preset-form>`.
```
or something similar, but I think it's worth it to be a bit more precise here? (and mention this is specific to presets, like it's implied in my suggestion)

> - Stages, i.e., the status such as `Ready`, highlighted in the defined colors.

Reviewer: This is only true if you don't click a specific stage in the top bar (i.e., if you're display "All" orders), should we briefly mention this here?

Author: I will mention it, but it is actually also common sense that you don't indicate the stage when you are already on this filtered stage page

Reviewer: Yeah, I agree. I was on the fence writing that comment; on one hand I was trying to be as accurate as possible, but you're also right that some things are common sense as well. Let's ignore my comment here then! :wink:

- Waiting time, with visual indicators, if the elapsed time exceeds the predefined alert time.

To update order progress from the order card, click individual items to cross them off individually
or click the order card itself to mark all items at once. Once all items are completed, the order
automatically moves to the next stage. Click :icon:`fa-undo` :guilabel:`Reset` on
:guilabel:`Completed` orders to move them back to the first stage if needed.

> .. example::

Reviewer: I think adding an concrete example here, with so many different use cases, is a very good idea, thank you!

Author: Thank you ❤️

   The preparation display shows several orders to process:

   - Order 1: The `Lunch Maki 18pc` for takeout with order name `John Meyer`, scheduled for `14/04`

>      at `13:00`, is ready (crossed out). The `Bacon Burger` is still in preparation.

Reviewer: ```suggestion
     at `13:00`, is ready (crossed out). The `Bacon Burger` is still being prepared.
```

   - Order 2: The delivery order for the customer `Acme Corporation` is marked red because it

>      exceeds the expected time in the `Ready` stage.

Reviewer: ```suggestion
    has exceeded the expected time in the `Ready` stage.
```

   - Order 3: Table 2 `T2` has a `Dine in` order with 3 dishes and 3 guests (:icon:`fa-users` icon)
     at this table. You can click `Done` to archive the order or :guilabel:`Reset` to move it back
     to the `To cook` stage.

   .. image:: preparation/preparation-display-cards.png
      :alt: The preparation display interface with orders to process.
      :scale: 80 %

.. _preparation/order-status-screen:

Using the order status screen
-----------------------------

The :guilabel:`Order Status Screen` is automatically enabled once a :ref:`preparation display

> <pos/preparation/application>` is created. It is an additional display to show customers an overview

Reviewer: ```suggestion
<pos/preparation/application>` is created. It is an additional display showing customers an overview
```
